expression/builtin_cast.go

```diff
@@ -159,29 +114,27 @@ func (c *castAsIntFunctionClass) getFunction(args []Expression, ctx context.Cont
 	if err := c.verifyArgs(args); err != nil {
 		return nil, errors.Trace(err)
 	}
-	bf := baseIntBuiltinFunc{newBaseBuiltinFunc(args, ctx)}
-	bf.tp = c.tp
```

Comment:
why not remain these two lines and set `bf.pbCode` before every `sig` constructed, then we do not need to implement a `new()` method for every `sig`